src/main/java/teammates/ui/webapi/CreateAccountAction.java

```diff
@@ -1,59 +1,99 @@
+    AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    void checkSpecificAccessControl() {
+        // Any user can create instructor account as long as the registration key is valid.
+    }
+
+    @Override
+    public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
+        String registrationKey = getNonNullRequestParamValue(Const.ParamsNames.REGKEY);
+
+        AccountRequestAttributes accountRequestAttributes = logic.getAccountRequestForRegistrationKey(registrationKey);
+
+        if (accountRequestAttributes == null) {
+            throw new EntityNotFoundException("Account request with registration key "
+                    + registrationKey + " could not be found");
+        }
+
+        if (accountRequestAttributes.getRegisteredAt() != null) {
+            throw new InvalidOperationException("The registration key " + registrationKey + " has already been used.");
+        }
+
+        String instructorEmail = accountRequestAttributes.getEmail();
+        String instructorName = accountRequestAttributes.getName();
+        String instructorInstitution = accountRequestAttributes.getInstitute();
 
-        String instructorName = createRequest.getInstructorName().trim();
-        String instructorEmail = createRequest.getInstructorEmail().trim();
-        String instructorInstitution = createRequest.getInstructorInstitution().trim();
         String courseId;
 
         try {
             courseId = importDemoData(instructorEmail, instructorName, instructorInstitution);
-        } catch (InvalidParametersException e) {
+        } catch (InvalidParametersException ipe) {
             // There should not be any invalid parameter here
-            log.severe("Unexpected error", e);
-            return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+            log.severe("Unexpected error", ipe);
+            return new JsonResult(ipe.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
+
         List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(courseId);
-        String joinLink = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
-                .withRegistrationKey(instructorList.get(0).getKey())
-                .withEntityType(Const.EntityType.INSTRUCTOR)
-                .toAbsoluteString();
-        EmailWrapper email = emailGenerator.generateNewInstructorAccountJoinEmail(
-                instructorList.get(0).getEmail(), instructorName, joinLink);
-        emailSender.sendEmail(email);
-
-        JoinLinkData output = new JoinLinkData(joinLink);
-        return new JsonResult(output);
+
+        try {
+            logic.joinCourseForInstructor(instructorList.get(0).getKey(), userInfo.id);
+        } catch (EntityDoesNotExistException ednee) {
+            throw new EntityNotFoundException(ednee);
```

Comment:
You may want to check whether all of the exceptions thrown here are possible under normal conditions as well. At least `EntityDoesNotExistException` is not for reasons quite obvious.

Reply:
I don't believe any of them are possible, will make changes.